pkg/kv/kvserver/store_snapshot.go

```diff
@@ -387,25 +436,49 @@ func (kvSS *kvBatchSnapshotStrategy) Send(
 			return 0, err
 		}
 
-		if bLen := int64(b.Len()); bLen >= kvSS.batchSize {
-			if err := kvSS.sendBatch(ctx, stream, b); err != nil {
+		if int64(b.Len()) >= kvSS.batchSize {
+			if err = flushBatch(); err != nil {
 				return 0, err
 			}
-			bytesSent += bLen
-			recordBytesSent(bLen)
-			b.Close()
-			b = nil
 		}
 	}
+	if err != nil {
+		return 0, err
+	}
+
+	iter = snap.RangeKeyIter
```

Comment:
Reuse of iter variable is confusing at first. While iter is close to 'i' as a index loop it has richer semantics and we seem to be reusing iter, but with different functionality.
It is not dissimilar to our use or range word which is heavily context dependent.